trl/trainer/grpo_trainer.py

```diff
@@ -836,13 +932,14 @@ def _generate_and_score_completions(
         attention_mask = torch.cat([prompt_mask, completion_mask], dim=1)  # (B, P+C)
 
         logits_to_keep = completion_ids.size(1)  # we only need to compute the logits for the completion tokens
+        batch_size = self.args.per_device_train_batch_size if mode == "train" else self.args.per_device_eval_batch_size
```

Comment:
Generally for training (independent of this PR) `per_device_eval_batch_size` is `2*per_device_train_batch_size`

As we are in `no_grad` mode you can probably set the `per_device_train_batch_size` here to be `2*self.args.per_device_train_batch_size` as there will be no memory used for the storage of activations etc.

It would be worth validating in a scenario with significant memory pressure, so do not worry about it if you want to get the PR merged.